packages/test/test-end-to-end-tests/src/test/gc/gcReferenceUpdatesInSummary.spec.ts

```diff
@@ -23,56 +17,58 @@ import { ITestObjectProvider, waitForContainerConnection } from "@fluidframework
 import { describeFullCompat } from "@fluid-internal/test-version-utils";
 import { UndoRedoStackManager } from "@fluidframework/undo-redo";
 
-class TestDataObject extends DataObject {
-	public get _root() {
-		return this.root;
-	}
+/**
+ * Validates this scenario: When all references to a data store are deleted, the data store is marked as unreferenced
+ * in the next summary. When a reference to the data store is re-added, it is marked as referenced in the next summary.
+ * Basically, if the handle to a data store is not stored in any DDS, its summary tree will have the "unreferenced"
+ * property set to true. If the handle to a data store exists or it's a root data store, its summary tree does not have
+ * the "unreferenced" property.
+ */
+describeFullCompat("GC reference updates in local summary", (getTestObjectProvider, apis) => {
+	class TestDataObject extends apis.dataRuntime.DataObject {
+		public get _root() {
+			return this.root;
+		}
 
-	public get _context() {
-		return this.context;
-	}
+		public get _context() {
+			return this.context;
+		}
 
-	private readonly matrixKey = "matrix";
-	public matrix!: SharedMatrix;
-	public undoRedoStackManager!: UndoRedoStackManager;
+		private readonly matrixKey = "matrix";
+		public matrix!: SharedMatrix;
```

Comment:
This also has to be versioned right. CompatApis should include DDSs as well.

Reply:
Wow, this logic is so complicated in the comapt utils. It looks like the code you shared only gets used when `makeTestContainer` or `loadTestContainer` are called and not when `createContainer` or `loadContainer` are called?

Reply:
ahh yeah you're right. So we would need it here anyway.

Reply:
So, what's the plan for this? Are you planning to do this in a follow up?

Reply:
no, it's in this PR. It didn't require any code updates in the test data object because I destructured `apis.dds` into `SharedMatrix` and `SharedString`, so that the object registry and the calls to create both refer to factories from whatever compat layer we're testing.

I think this is a bit more clear if I move that destructuring up to the start of the block though, so I did that.

Reply:
Oh I see. I totally missed that change!